### DISCUSSION TITLE
My grab bag of desired future functionality

### DISCUSSION THREAD

**Discussion by github-wlk** [hydrogen-music/hydrogen#1129] [Ideas]
Hi, I use hydrogen on Linux, and here are some ideas that I would like to see.  I wrote them down for myself, but I might as well annoy a bunch of others :)

1) Drumkit variants.   New "drumkit" but based on a more "fixed" set of source files (wav), in order to save space and make things more flexible (see some of the following points).   So, let's say you have drumkit X, and you want to save a version of it which is using a different midi channel (just as an example... don't argue yet :))  It could be a different drumkit.xml file, or a different stanza in the "standard" kit's drumkit.xml file.   I'll let you argue the merits of those options.  I would just like to see more flexibility in having various versions and tweaks of a certain drumkit, without having the overhead of saving a whole new copy of the wavs.

2) Tuning.  Mass tuning, perhaps, to be more precise.   What I'm imagining...   A drumkit is selected.  You pull up a window which will list all the instruments in that drumkit, along with the tuning.  A dial or field to modify all the tuning for that drumkit.  Checkboxes next to each instrument to determine whether you want that tuning change to apply (you want to change the pitch of the toms, but not the cymbals, etc).   You see what I'm going at....   A master drum-tuning page.   Then, you can save that drumkit as a variant (So you'd have the original drumkit X, and now also, say, "X in Bb".  Same set of wavs.

3) Midi channel.   See (2) above...  change the midi channel for all or select instruments, and then save it as a variant.  Long story short, we'd have a master drumkit editor where one could manipulate the options more easily.   The current UI is fine, for each individual instrument, but to have to go through and change the tuning on every sample for every instrument using the current situation.... that's a bear.

Thank you for letting me vent :)   No, I don't know how other (commercial) products work, so I'm not comparing this to someone else.   My personal use scenario:   I use Ardour, and I like H2's humanizing options.   So I prefer to write my percussion parts in H2, and then play and print them into Ardour.  I guess, in addition to the changes I recommend above, having H2 as an LV2 plugin may be helpful, or even to have Ardour develop into more of a framework, allowing one to choose from foss and commercial "plugins" to handle things such as midi, etc.   But that is not your problem :)   With my use scenario, for example, I could take a drumset like the AVL RedZep, for example, be able to save various variants based on tuning, and also have variants where it would output sound through H2, and perhaps another variant where I'm outputting miidi during playback....   So I can either print audio or midi in Ardour.   A boy can dream.  Thanks again!

**Comment by trebmuh** [comment]
Hi and, first of all, thanks for the suggestions.

That being said, they would probably be better separated in 3 different threads with much better names (aka "`My grab bag of desired future functionality`" only make sense for you, not for all the other people reading the thread). Could you please do so?

**Comment by github-wlk** [comment]
created https://github.com/hydrogen-music/hydrogen/discussions/1134 and https://github.com/hydrogen-music/hydrogen/discussions/1136

**Reply by trebmuh** [reply]
@github-wlk thank you very much for that.
Is that OK fro you to close this thread now?

**Comment by github-wlk** [comment]
No problem.   Close away.
